### DISCUSSION TITLE
Login - Invalid uname / pword

### DISCUSSION THREAD

**Discussion by NotReallyNotNotMe** [YOURLS/YOURLS#3255] [Q&A]
Ref: https://github.com/YOURLS/YOURLS/discussions/3254
Following on discussion with dgw - After one successful login, I am unable to login. Again, I'm getting the listing of folders under bronord.com/yourls. If I clidk on the admin folder, the login prompt appears and logins fail. I see no error logs in any part of the tree. 
I have edited config.php to change username and password. I even created a new uname/pword pair. When I look at config.php on the server, the changes I made are present but the pword is not encrypted. Next step??

**Comment by dgw** [comment]
The next step when you find a problem should always be [troubleshooting](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps)—in particular, enabling debug mode. There's no substitute for server logs. We aren't psychic. 😅

**Reply by NotReallyNotNotMe** [reply]
No error logs that I can access for web server. I turned on debugging in config.php and get the following:
Connected to database bronordc_yourls on localhost;port=3306
SQL perform: SELECT option_name, option_value FROM yourls_rogeroptions WHERE 1=1 (0.00034 s)
cURL error 60: SSL certificate problem: certificate has expired (POST on https://api.yourls.org/core/version/1.0/)
SQL perform: UPDATE yourls_rogeroptions SET option_value = 'O:8:"stdClass":4:{s:15:"failed_attempts";i:50;s:12:"last_attempt";i:1647547163;s:11:"last_result";s:0:"";s:15:"version_checked";s:5:"1.8.2";}' WHERE option_name = 'core_version_checks' (0.00549 s)
Check for new version: no

I checked with my hosting service and my SSL is up-to-date. (If it was not, I probably could not use ssh and get to bronord.com).

Enlightenment is desired. /Roger

**Reply by dgw** [reply]
SSL and SSH are unrelated, despite the similar acronyms. At any rate, that error is about something on the YOURLS version check service (which I can't reproduce on my own system, but let's move on).

You see a directory listing at `/yourls` because there is no index file in that folder. See [the FAQ entry](https://yourls.org/#FAQ;noindex), or look for LiteSpeed documentation on how to disable the `autoIndex` option. You may have to contact your host to do it, depending on how they have it set up.

If YOURLS isn't able to encrypt the plaintext passwords in its config file, there should be an error somewhere. It might only appear once you successfully authenticate, however. Failed logins and failed password encryption could both result from an invalid credential array. Are you certain that the format matches the examples in [the wiki page](https://github.com/YOURLS/YOURLS/wiki/Username-Passwords#editing-login--passwords-in-configphp)?

**Reply by NotReallyNotNotMe** [reply]
I'll check these things tomorrow but am close to giving up / trying to learn new things with an olde brain! Thanks for engaging on this.

**Reply by NotReallyNotNotMe** [reply]
I checked with my hosting service and they point to a cert problem at api.yourls.org:

. . . in regards to your error - I am seeing that api.yourls.org is still using the X3 authority in its certificate:
[Issuer: DST Root CA X3] https://www.sslshopper.com/ssl-checker.html#hostname=api.yourls.org

This CA was expired/discontinued in September, 2021 -> https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/

While the certificate will test on SSLShopper.com as 'valid' it is generally un-trusted by the majority of the internet - us included. It's signed by a certificate authority certificate that has essentially been revoked.

It will be up to whoever controls api.yourls.org to resolve this - it's not something that we can fix on our end.

**Reply by LeoColomb** [reply]
Not only their answer is false (the certificate is using ISRG Root X1 as root certificate, [which is 100% valid](https://www.ssllabs.com/ssltest/analyze.html?d=api.yourls.org), especially because the certificate is renewed every quarter so it can't used un deprecated root one), but this is completely unrelated to the main error you are facing.

**Reply by dgw** [reply]
@LeoColomb If you expand the certification paths at the SSLLabs link you gave, Léo, you'll see a second chain that goes through an alternate (cross-signed) ISRG X1 root to the expired R3 cert. The host's statement isn't exactly false, but it indicates they are possibly using outdated libs/cacerts. OpenSSL changed how it handles multiple possible chains. https://www.openssl.org/blog/blog/2021/09/13/LetsEncryptRootCertExpire/

It is entirely unrelated to the OP's login issue, though, that's true.

**Reply by LeoColomb** [reply]
Indeed, you are right, not exactly false but quite off the mark 😄

**Reply by NotReallyNotNotMe** [reply]
1. There IS an index.html in /admin.
2. A copy of my config.php is in the attached zip file, with fudged uname and pword. Do you mind taking a look at it? The index.html from /yourls is also therein.
3. So, what am I left with? What to check?
4. [problems.zip](https://github.com/YOURLS/YOURLS/files/8295004/problems.zip)

Thanks for puzzling over this.

**Reply by dgw** [reply]
> 1. There IS an index.html in /admin.

It's `index.php`, and that was never in question. It clearly works, since the login form appears.

> 2. A copy of my config.php is in the attached zip file, with fudged uname and pword. Do you mind taking a look at it? The index.html from /yourls is also therein.

I'll address the fudged config below. Here again, that is `index.php`, not `index.html`. It's clearly the one from `/yourls/admin/index.php`, which ships with YOURLS.

> 3. So, what am I left with? What to check?

I realize now that you've not given any detail about the problem with authentication beyond "logins fail". And if YOURLS is unable to hash the passwords, you won't see that error message until you can successfully log in.

Honestly, I'm not sure what's happening. Nothing looks wrong in the fudged config. I've not used the short `[]` array syntax myself, but there's no reason that should break YOURLS. You could try replacing it with the older `array()` style, just to see if it helps? "Try everything" is often a good troubleshooting approach.

The good news is that redirections appear to work. The sample links added by YOURLS during install (e.g. https://bronord.com/yourls/ozh) function as expected.

**Important security considerations:** Don't forget to reset your `YOURLS_COOKIEKEY` and `YOURLS_DB_PASS`, unless you fudged those too. Also please do remember to delete the `YOURLS_DEBUG` line you added near the top of the config when done troubleshooting.

**Comment by NotReallyNotNotMe** [comment]
On 3/17/22 20:01, dgw wrote:


   1. There IS an index.html in /admin.

It's index.php, and that was never in question. It clearly works, since the
login form appears.


   1. A copy of my config.php is in the attached zip file, with fudged
   uname and pword. Do you mind taking a look at it? The index.html from
   /yourls is also therein.

I'll address the fudged config below. Here again, that is index.php, not
index.html. It's clearly the one from /yourls/admin/index.php, which ships
with YOURLS.

I write my web pages in "raw" html. I think and even dream in html, hence
the slip of the tongue. Maybe I need to see a shrink about this.


   1. So, what am I left with? What to check?

I realize now that you've not given any detail about the problem with
authentication beyond "logins fail". And if YOURLS is unable to hash the
passwords, you won't see that error message until you can successfully log
in.

Honestly, I'm not sure what's happening. Nothing looks wrong in the fudged
config. I've not used the short [] array syntax myself, but there's no
reason that should break YOURLS. You could try replacing it with the older
array() style, just to see if it helps? "Try everything" is often a good
troubleshooting approach.

I'm willing to give the "older array" style a try. But how?

Trying everything is good. It's te Linux way: lottsa self flagellation.

The good news is that redirections appear to work. The sample links added
by YOURLS during install (e.g. https://bronord.com/yourls/ozh) function as
expected.

Good to know but I don't really know about the gutz of yourls so this
doesn't mean as much to me as to you.

*Important security considerations:* Don't forget to reset your
YOURLS_COOKIEKEY and YOURLS_DB_PASS, unless you fudged those too. Also
please do remember to delete the YOURLS_DEBUG line you added near the top
of the config when done troubleshooting.

Good point. Will delete the debug line when success is attained and change
codes.

Thanks again.

On Thu, Mar 17, 2022 at 8:01 PM dgw ***@***.***> wrote:

>
>    1. There IS an index.html in /admin.
>
> It's index.php, and that was never in question. It clearly works, since
> the login form appears.
>
>
>    1. A copy of my config.php is in the attached zip file, with fudged
>    uname and pword. Do you mind taking a look at it? The index.html from
>    /yourls is also therein.
>
> I'll address the fudged config below. Here again, that is index.php, not
> index.html. It's clearly the one from /yourls/admin/index.php, which
> ships with YOURLS.
>
>
>    1. So, what am I left with? What to check?
>
> I realize now that you've not given any detail about the problem with
> authentication beyond "logins fail". And if YOURLS is unable to hash the
> passwords, you won't see that error message until you can successfully log
> in.
>
> Honestly, I'm not sure what's happening. Nothing looks wrong in the fudged
> config. I've not used the short [] array syntax myself, but there's no
> reason that should break YOURLS. You could try replacing it with the older
> array() style, just to see if it helps? "Try everything" is often a good
> troubleshooting approach.
>
> The good news is that redirections appear to work. The sample links added
> by YOURLS during install (e.g. https://bronord.com/yourls/ozh) function
> as expected.
>
> *Important security considerations:* Don't forget to reset your
> YOURLS_COOKIEKEY and YOURLS_DB_PASS, unless you fudged those too. Also
> please do remember to delete the YOURLS_DEBUG line you added near the top
> of the config when done troubleshooting.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/YOURLS/YOURLS/discussions/3255#discussioncomment-2384280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH5KIUQXQON4VWYZ4YKOYDLVAPBVNANCNFSM5Q7UKP4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Roger

**Reply by NotReallyNotNotMe** [reply]
I am flumoxed! Probably an error on my end. I reviewed config.php and updated it. I totally deleted software from my host and re-installed. The messages said the install was successful. Connection to the DB seemed to "go." BUT, I am still getting login failures. I have read and re-read the instructions but may not be understanding them. Is there an example with "real" / example user data that I can look at, in case I'm not getting the proper username / pword? I am lost about where to go to see why I am having this difficulty. Please note: I'm an end user and not a developer, so my skills are a bit limited. Thanks for looking at this and your patience. /Roger

**Reply by dgw** [reply]
> BUT, I am still getting login failures.

BUT, you still have not provided the error message or other feedback YOURLS gives you when login fails.

> Is there an example with "real" / example user data that I can look at, in case I'm not getting the proper username / pword?

You mean like [the example in the wiki page previously linked](https://github.com/YOURLS/YOURLS/wiki/Username-Passwords#editing-login--passwords-in-configphp), which you could paste directly into your `config.php` file and then try logging in with one or more of the example accounts?

**Reply by NotReallyNotNotMe** [reply]
Giving up. Lost in cyberspace! Thanks for trying to help.
